### PR TITLE
Fix MSVC errors

### PR DESCRIPTION
**Pull request by shohirose**
I made several changes to your library. I hope these changes improve your code.

The reasons of the 2nd commit about /utf-8 and /EHsc are
- Because compiling lexy with MSVC requires utf-8, it is natural to add /utf-8 option to lexy_dev itself.
- When opening the root directory with Visual Studio, VS does not automatically add /EHsc option, leading to a compilation error. (This does not occur when using VSCode.)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,12 @@ docs/content/learn/changelog.adoc
 docs/static/download/*.zip
 docs/public/
 docs/resources/
+
+# CMake
```
Comment (foonathan) [inline .gitignore:8]: I don't want to add user specific stuff to .gitignore. Those belong in your user global .gitignore or .git/ignore. The project .gitignore contains only files that are generated by the project's build system.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -191,7 +191,7 @@ elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
     # GCC's arry bounds, maybe uninitialized, and restrict warning seems to have false positives.
     target_compile_options(lexy_dev INTERFACE -Wno-array-bounds -Wno-maybe-uninitialized -Wno-restrict)
 elseif(MSVC)
-    target_compile_options(lexy_dev INTERFACE /WX /W3 /D _CRT_SECURE_NO_WARNINGS /wd5105)
+    target_compile_options(lexy_dev INTERFACE /WX /W3 /D _CRT_SECURE_NO_WARNINGS /wd5105 /EHsc /utf-8)
```
Comment (foonathan) [inline src/CMakeLists.txt:194]: What is the error /EHsc fixes?

File: tests/lexy/detail/string_view.cpp [hunk 3]
```diff
@@ -39,7 +39,7 @@ constexpr auto fn()
 
 TEST_CASE("make_cstr")
 {
-    constexpr auto str = lexy::_detail::make_cstr<+fn>;
+    constexpr auto str = lexy::_detail::make_cstr<&fn>;
```
Comment (foonathan) [inline tests/lexy/detail/string_view.cpp:42]: It did not compile before?

**Review by foonathan — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (shohirose) [reply .gitignore:8]: I got it.

File: src/CMakeLists.txt (hunk 2 above)
Comment (shohirose) [reply src/CMakeLists.txt:194]: The corresponding errror message is:

```
[5/301] Building CXX object examples\CMakeFiles\lexy_example_protobuf.dir\protobuf.cpp.obj
FAILED: examples/CMakeFiles/lexy_example_protobuf.dir/protobuf.cpp.obj 
C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1437~1.328\bin\Hostx64\x64\cl.exe  /nologo /TP 
    -IC:\Users\shohi\Documents\github\lexy\src\..\include /utf-8 /MDd /Zi /Ob0 /Od /RTC1 -std:c++20 /WX /W3 /D 
    _CRT_SECURE_NO_WARNINGS /wd5105 /utf-8 
    /showIncludes /Foexamples\CMakeFiles\lexy_example_protobuf.dir\protobuf.cpp.obj 
    /Fdexamples\CMakeFiles\lexy_example_protobuf.dir\ /FS -c C:\Users\shohi\Documents\github\lexy\examples\protobuf.cpp
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\vector(835): error C2220: warning treated as error - no object file generated
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\vector(835): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
```

File: tests/lexy/detail/string_view.cpp (hunk 3 above)
Comment (shohirose) [reply tests/lexy/detail/string_view.cpp:42]: No, it didn't. The error message was:

```
[1/27] Building CXX object tests\lexy\CMakeFiles\lexy_test.dir\detail\string_view.cpp.obj
FAILED: tests/lexy/CMakeFiles/lexy_test.dir/detail/string_view.cpp.obj 
C:\PROGRA~1\MIB055~1\2022\COMMUN~1\VC\Tools\MSVC\1437~1.328\bin\Hostx64\x64\cl.exe  /nologo /TP -DLEXY_HAS_UNICODE_DATABASE=1 
    -DLEXY_TEST -IC:\Users\shohi\Documents\github\lexy\src\..\include 
    -IC:\Users\shohi\Documents\github\lexy\out\build\x64-Debug\_deps\doctest-src /utf-8 /MDd /Zi /Ob0 /Od /RTC1 
    -std:c++20 /Zc:preprocessor /utf-8 /bigobj /WX /W3 /D _CRT_SECURE_NO_WARNINGS /wd5105 /EHsc 
    /showIncludes /Fotests\lexy\CMakeFiles\lexy_test.dir\detail\string_view.cpp.obj 
    /Fdtests\lexy\CMakeFiles\lexy_test.dir\ 
    /FS -c C:\Users\shohi\Documents\github\lexy\tests\lexy\detail\string_view.cpp
C:\Users\shohi\Documents\github\lexy\tests\lexy\detail\string_view.cpp(42): error C2975: 'FnPtr': invalid template argument for 'lexy::_detail::make_cstr', expected compile-time constant expression
```

File: src/CMakeLists.txt (hunk 2 above)
Comment (foonathan) [reply src/CMakeLists.txt:194]: Hm, I have a hard time believing that this is default CMake behavior. All sources I could find point to "/EHsc" being applied by default. Could this apply to you perhaps? https://github.com/microsoft/verona/issues/49#issuecomment-576457428

Comment (shohirose) [reply src/CMakeLists.txt:194]: It seems relevant to my problem.
Altough the error only occured when opening the project using Visual Studio, it suddenly dissapeared after I updated VS from 17.7.5 to 17.7.6 a few days ago... strange.
I think we do not have to put /EHsc option now as you mentioned.

Comment (foonathan) [reply src/CMakeLists.txt:194]: Okay. Can you do a force push with two commits: One that fixes the typo and one thats add /utf-8?

**Comment by foonathan** [comment]
Thank you!

**Pull request merged**